uaclient/status.py

```diff
@@ -1,14 +1,53 @@
+import enum
+
+
 class TxtColor:
     OKGREEN = '\033[92m'
     DISABLEGREY = '\033[37m'
     FAIL = '\033[91m'
     ENDC = '\033[0m'
 
 
+@enum.unique
+class ApplicationStatus(enum.Enum):
+    """
+    An enum to represent the current application status of an entitlement
+    """
+    ENABLED = object()
+    DISABLED = object()
+
+    def to_operational_status(self) -> str:
+        mapping = {
+            ApplicationStatus.ENABLED: ACTIVE,
+            ApplicationStatus.DISABLED: INACTIVE,
+        }
+        return mapping[self]
+
+
+@enum.unique
+class ContractStatus(enum.Enum):
+    """
+    An enum to represent whether a user is entitled to an entitlement
+
+    (The value of each member is the string that will be used in status
+    output.)
+    """
+    ENTITLED = 'entitled'
+    UNENTITLED = 'none'
+
+
+@enum.unique
+class ApplicabilityStatus(enum.Enum):
+    """
+    An enum to represent whether an entitlement could apply to this machine
+    """
+    APPLICABLE = object()
+    INAPPLICABLE = object()
+
+
 ACTIVE = 'active'
```

Comment:
Can we drop ACTIVE and INACTIVE now and use the ApplicationStatus enum as the reference for those static stings?

Reply:
As I mentioned above, something will happen to ACTIVE/INACTIVE once `operational_status()` is gone, but it's not worth refactoring them before that.